### PR TITLE
Fixed small issues, added connection listen code.

### PR DESCRIPTION
**Pull request by Revertron**
### Description of the Change

* Fixed issues from https://github.com/hyperledger/iroha/pull/1083
* Added connection listening code.

### CODE REVIEW

File: iroha_p2p/src/network.rs [hunk 1]
```diff
@@ -31,18 +35,27 @@ where
     E: Encryptor + Send + 'static,
 {
     /// Creates a network structure, that will hold connections to other nodes.
-    pub fn new(broker: Broker /*handler: AlwaysAddr<A>*/) -> Self {
-        Self {
+    pub fn new(
+        broker: Broker,
+        listen_addr: String, /*handler: AlwaysAddr<A>*/
```
Comment (rkharisov) [inline iroha_p2p/src/network.rs:40]: Remove commented code?

**Review by rkharisov — DISMISSED** [review]

File: iroha_p2p/src/network.rs [hunk 2]
```diff
@@ -31,18 +35,27 @@ where
...
+    ) -> Result<Self, Error> {
+        let listen_addr: SocketAddr = listen_addr.parse()?;
+        Ok(Self {
             peers: HashMap::new(),
+            listen_addr,
```
Comment (i1i1) [inline iroha_p2p/src/network.rs:45]: You definitely can open listener here. Address is not state of network

File: iroha_p2p/src/network.rs [hunk 3]
```diff
@@ -61,6 +74,14 @@ where
         }
     }
 
+    /// Creates a [`TcpListener`] to listen to external connections.
+    async fn listen(&mut self) -> Result<(), Error> {
+        let listener = TcpListener::bind(self.listen_addr).await?;
+        self.listener = Some(listener);
```
Comment (i1i1) [inline iroha_p2p/src/network.rs:80]: It should be part of constructor

File: iroha_p2p/src/network.rs [hunk 4]
```diff
@@ -86,8 +107,17 @@ where
...
-        ctx.notify_with_context(self.listener_stream());
+        // Start listener
+        match self.listen().await {
+            Ok(_) => {
+                // register for peers from listener
+                ctx.notify_with_context(self.listener_stream());
+            }
+            Err(e) => {
+                // TODO send some message to Sumeragi that we couldn't start
+                warn!(%e, "Error starting listener!");
```
Comment (i1i1) [inline iroha_p2p/src/network.rs:118]: That is an error though

File: iroha_p2p/src/lib.rs [hunk 5]
```diff
@@ -27,6 +27,15 @@ pub enum Error {
...
     Keys,
+    /// Failed to parse address
+    #[error("Failed to parse socket address")]
+    Addr,
+}
+
+impl From<AddrParseError> for Error {
+    fn from(_: AddrParseError) -> Self {
+        Self::Addr
+    }
```
Comment (i1i1) [inline iroha_p2p/src/lib.rs:38]: Is there a reason to discard addr parse error?

**Review by i1i1 — CHANGES_REQUESTED** [review]

File: iroha_p2p/src/network.rs (hunk 1 above)
Comment (Revertron) [reply iroha_p2p/src/network.rs]: Removed. Will ad it later :)

File: iroha_p2p/src/network.rs (hunk 2 above)
Comment (Revertron) [reply iroha_p2p/src/network.rs]: Moved TcpListener start to constructor.

File: iroha_p2p/src/network.rs (hunk 4 above)
Comment (Revertron) [reply iroha_p2p/src/network.rs]: Moved TcpListener start to constructor. Error will bubble up.

File: iroha_p2p/src/lib.rs (hunk 5 above)
Comment (Revertron) [reply iroha_p2p/src/lib.rs]: Changed the type of `Error::Addr`, removed this impl.

**Review by i1i1 — APPROVED** [review]

**Review by i1i1 — APPROVED** [review]

**Review by rkharisov — APPROVED** [review]

**Review by eadventurous — APPROVED** [review]

**Pull request merged**